### ISSUE TITLE
DLSS ?

### ISSUE THREAD

**Issue by Appolon24800** [xCollateral/VulkanMod#556]
### Describe the request

Hey, this question might be dumb but I know that Nvidia doesn't support DLSS for OpenGL but does support  Vulkan so would it be possible to have DLSS in Minecraft java with your mod ?

**Comment by aliceles0** [comment]
why would you need dlss though

**Comment by terraquad** [comment]
> why would you need dlss though

better performance since the game runs on a lower resolution internally
but this probably won't be supported by the mod because this is a vendor specific extension
and this mod focuses on features supported by most major GPU manufacturers

**Comment by aliceles0** [comment]
why would need dlss in a block game though its not like a triple a game

**Comment by aliceles0** [comment]
also dlss doesnt just have the game on a lower resolution im pretty sure it lowers the resolution then upscales it, idk why you would need that for minecraft

**Comment by terraquad** [comment]
> also dlss doesnt just have the game on a lower resolution im pretty sure it lowers the resolution then upscales it, idk why you would need that for minecraft 

It would still have better performance since the shaders (the ones included by minecraft) would run a lot less times,
but this doesn't do much because minecrafts shaders aren't that complex compared to other games

**Comment by Phiro811** [comment]
I mean who doesnt wants to have more FPS.

**Comment by aliceles0** [comment]
i wouldnt really want to reduce my resolution in a block game though

**Comment by terraquad** [comment]
> i wouldnt really want to reduce my resolution in a block game though

It renders at a low resolution and upscales the produced image to full quality using AI (see https://en.wikipedia.org/wiki/Deep_Learning_Super_Sampling)

This would technically boost the game FPS since the shaders don't have to run for each pixel on the screen, but as stated above, it doesn't really matter in a game like Minecraft and this mod doesn't implement vendor-specific features.

> Edit: I didn't realize at the time that the upscaler has a certain performance cost itself...

**Comment by GrimPlayzs** [comment]
Just why?

DLSS is not a priority nor should it be, and will have little to no benefit in a voxel-based block game.
You would want to run the game, or any game, ideally at the native resolution of your monitor. Technologies such as DLSS and FSR exist to improve performance in highly demanding titles for those who might not have the top-of-the-line GPU. 

We should not rely on such technologies to make a title playable. The game industry already relies on these technologies too much. Look at any modern title and turn off DLSS/FSR and they run like crap.

**Comment by MrSzalonna** [comment]
> Just why?
> 
> DLSS is not a priority nor should it be, and will have little to no benefit in a voxel-based block game. You would want to run the game, or any game, ideally at the native resolution of your monitor. Technologies such as DLSS and FSR exist to improve performance in highly demanding titles for those who might not have the top-of-the-line GPU.
> 
> We should not rely on such technologies to make a title playable. The game industry already relies on these technologies too much. Look at any modern title and turn off DLSS/FSR and they run like crap.

It would be nice. And you could just use dlaa to improve visual quality without sacrificing frames. And if you have a 2050 and you want to play with 48 or 64 chunks. I play with 96 and runs fine on a 4080.

**Comment by MrSzalonna** [comment]
It would be good for me to play with anti aliasing. Dlss has a function called dlaa, which make the game run on a higher resolution using dlss. It would improve visual quality. Minecraft runs fine on my pc, but the poor people who are stuck with a 2050 laptop, it suck on higher render distances. Dlss is open source I think. Also if we have Vulkan, why wouldn't we start testing things like ray tracing and dlss? Even fsr and xess. Fsr would improve the fps for integrated graphics and xess for low end intel dedicated graphics.

**Comment by aliceles0** [comment]
what is a 2050 laptop

**Comment by MrSzalonna** [comment]
A laptop computer with an Nvidia GeForce RTX 2050 Graphics Processing Unit. It's literally e-waste. https://www.techpowerup.com/gpu-specs/geforce-rtx-2050-mobile.c3859

**Comment by aliceles0** [comment]
i didnt know that was a thing

**Comment by Kabir404** [comment]
I feel like instead of DLSS, FSR2 should be implemented... or at least the first version, as it works nearly on every GPU hardware supporting Vulkan. 

A good implementation can be decent but it takes a bit of work for sure though.

**Comment by Appolon24800** [comment]
DLSS Could improve by A LOT Minecraft performance, modern GPUs are focused on DLSS and AI features (just see the RTX 50XX serie). I will be honest, I don't even know if its possible or not, if Nvidia (or AMD) has open sourced all the DLSS stuff but frame generation, up scalling etc.. (when I talk abt DLSS I mean all the AI features).

**Comment by Vavency** [comment]
> DLSS Could improve by A LOT Minecraft performance

No, it wouldn't as minecraft isn't even bottlenecked by the GPU but the data transfer between the CPU and GPU. FSR2/DLSS would do nothing to fix this while introducing currently unneeded complexity. Also vulkanmod is still in beta albeit very usable, other possible improvements could be done way before then. 

tl;dr it's not needed while dev wasting time

**Comment by FluffyPuppyKasey** [comment]
> It would be good for me to play with anti aliasing. Dlss has a function called dlaa, which make the game run on a higher resolution using dlss. It would improve visual quality. Minecraft runs fine on my pc, but the poor people who are stuck with a 2050 laptop, it suck on higher render distances. Dlss is open source I think. Also if we have Vulkan, why wouldn't we start testing things like ray tracing and dlss? Even fsr and xess. Fsr would improve the fps for integrated graphics and xess for low end intel dedicated graphics.

My integrated Intel Iris Xe graphics easily gets over 150FPS. Why would we need DLSS/FSR? Minecraft with VulkanMod runs on a damn potato

**Comment by Aberro** [comment]
I would say - having DLSS would be really nice, even for (mostly) vanilla, as DLSS, or rather DLAA, provides the best antialiasing there is. Better than supersampling in many cases, as supersampling is still prone to moire effect, whereas DLAA much less so.
But the main benefit of using it would be with heavy shaders, like Rethinking Voxels, which, on high resolution might make even RTX 5090 strain a lot to keep at 60fps.

The whole reason why I came up in this topic is that I was checking if maybe there's an option for some upscaling, like at least FSR, since currently, with Rethinking Voxels and Distant Horizons, at medium to high settings, the game often struggles to keep above 30 (which for me is ok, but still leaves much to be desired) on my RTX 3090. And it's definitely not CPU limited, and heavily resolution-dependent.

However, it's too early for this mod to actually consider implementing DLSS at this point, since there's practically no shaders yet, all shader injecting mods are OpenGL dependent and do not even plan to support VulkanMod (as it would require rewriting a mod from the ground), DH mod is OpenGL dependent too, and antialiasing alone simply do not worth the hassle. It might become a valid consideration at some distant future, if/when Vulkan would have much broader support/compatibility with existing shaders, but at this point, the whole discussion doesn't make much sense.

**Comment by NewGrafon** [comment]
DLAA (DLSS with 100% to 100% resolution) is great antialiasing, why not

**Comment by Noellelivinggayness** [comment]
fsr and xess makes way more sense since they work on anything and xess is so close to dlss you wouldn't be able to tell

**Comment by Noellelivinggayness** [comment]
also this is probably doable in a mod requiring this one

**Comment by MotWnb** [comment]
How about DLAA?

**Comment by blankuserrr** [comment]
> How about DLAA?

That is DLSS.

**Comment by Noellelivinggayness** [comment]
> > How about DLAA?
> 
> That is DLSS.

same tech different use case dlss would make absolutly no sense at all for minecraft unless you use heavy shaders and I am certain shaders can add FSR1 directly into the shader and they might be able to use other upscalers without even requiring vulkanmod to support it so it's already possible to implement in places where it's needed and without shaders dlss is completely useless at improving fps it will only lower fps by adding more complexity while optimizing something that was not even holding you back in the first place now using dlss as anti-aliasing is already more logical but guess what I've literally already seen upscaliers being implemented as anti-aliasing in shaders before and it's definitely even more possible with vulkanmod compatible shaders so again not their problem

**Comment by blankuserrr** [comment]
> same tech different use case dlss would make absolutly no sense at all for minecraft unless you use heavy shaders and I am certain shaders can add FSR1 directly into the shader and they might be able to use other upscalers without even requiring vulkanmod to support it so it's already possible to implement in places where it's needed and without shaders dlss is completely useless at improving fps it will only lower fps by adding more complexity while optimizing something that was not even holding you back in the first place now using dlss as anti-aliasing is already more logical but guess what I've literally already seen upscaliers being implemented as anti-aliasing in shaders before and it's definitely even more possible with vulkanmod compatible shaders so again not their problem

DLSS along with other upscalers does indeed make sense for low end systems and most shaders (since we don't have a visual-focused shader that doesn't reduce performance drastically yet). FSR 1-3 were quite bad as, even if they can be implemented in shaders, the only other one that can be implemented without VKM support is Snapdragon's which is based on FSR and actually looks worse due to focusing on mobile devices. You could implement TAAU but it's WIDELY hated. 

> dlss is completely useless at improving fps it will only lower fps by adding more complexity while optimizing something that was not even holding you back in the first place

I have no idea what you're talking about. Implementing DLSS doesn't add that much complexity, and ignoring shadows/AO/GI, rasterization inherently benefits from lower resolutions due to how it scales. 

> now using dlss as anti-aliasing is already more logical but guess what I've literally already seen upscaliers being implemented as anti-aliasing in shaders before and it's definitely even more possible with vulkanmod compatible shaders so again not their problem

Using DLSS would give you DLAA, it's not an either or; VulkanMod-compatible shaders are unofficial, spread across at least two different version targets, and implementing AA directly into the renderer will always be faster than post-processing.

**Comment by Noellelivinggayness** [comment]
> DLSS along with other upscalers does indeed make sense for low end systems and most shaders (since we don't have a visual-focused shader that doesn't reduce performance drastically yet). FSR 1-3 were quite bad as, even if they can be implemented in shaders, the only other one that can be implemented without VKM support is Snapdragon's which is based on FSR and actually looks worse due to focusing on mobile devices. You could implement TAAU but it's WIDELY hated.
> 
> I have no idea what you're talking about. Implementing DLSS doesn't add that much complexity, and ignoring shadows/AO/GI, rasterization inherently benefits from lower resolutions due to how it scales.
> 
> Using DLSS would give you DLAA, it's not an either or; VulkanMod-compatible shaders are unofficial, spread across at least two different version targets, and implementing AA directly into the renderer will always be faster than post-processing.

DLSS specifically would be stupid either ways the best thing you could maybe see happen is XeSS (which for me is the closest to DLSS in terms of visual quality) because it runs on most graphics card

and for lower end devices my shitbox workstation from 2007-2011 that was considered bad already at the time which can't even run this mod can run minecraft very well with the default settings and this is true even for 1.20.1 (last version I played on it)

if you can even run this mod you won't need upscaling unless you're using shaders and if you're using shaders then the shaders mod can (and it would make more sense) implement upscaling

the scope of this project is to optimize minecraft using vulkan upscaling is NOT optimization it's a trick for edge cases it makes no sense to ask vulkanmod to support upscaling
